doc/api/assert.md

```diff
@@ -633,13 +633,12 @@ changes:
 > functions instead.
 
 If `message` is falsy, the error message is set as the values of `actual` and
-`expected` separated by the provided `operator`. If just the two `actual` and
-`expected` arguments are provided, `operator` will default to `'!='`. If
-`message` is provided as third argument it will be used as the error message and
-the other arguments will be stored as properties on the thrown object. If
-`stackStartFn` is provided, all stack frames above that function will be
-removed from stacktrace (see [`Error.captureStackTrace`]). If no arguments are
-given, the default message `Failed` will be used.
+`expected` separated by the provided `operator`. If `operator` is falsy and no
+message is set, it will default to `'!='`. If `message` is truthy, the `operator`
+will default to `'fail'` and `message` will be used for the error message. Other
+arguments will be stored as properties on the thrown object. If `stackStartFn`
+is provided, all stack frames above that function will be removed from
+stacktrace (see [`Error.captureStackTrace`][]).
```

Comment:
I suggest to change the `operator` description above to explain when it's set to `'fail'` and when to `'!='`.

I would also move the `stackStartFn` description to the part above. That way the total description below concentrates on the general behavior instead of describing each individual option setting.

Right now we also miss to document that `assert.fail(1, 2, new Error()` throws the passed through error. That could be described in the `message` property above.